widget/entry-with-completion.go

```diff
@@ -0,0 +1,141 @@
+package widget
+
+import (
+	"fyne.io/fyne/v2"
+	"fyne.io/fyne/v2/widget"
+)
+
+// CompletionEntry is an Entry with options displayed in a PopUpMenu.
+//
+// Example:
+//
+//    entry := NewCompletionEntry()
+//    entry.OnChanged = func(s string) {
+//        if len(s) < 3 {
+//            entry.HideCompletion()
+//        }
+//        entry.SetItems([]string{"foo", "bar", "baz"}
+//        entry.ShowCompletion()
+//    }
+type CompletionEntry struct {
+	*widget.Entry
+	menu  *fyne.Menu
+	cmenu *widget.PopUpMenu
+	items []string
+	pause bool
+}
+
+// NewCompletionEntry creates a new CompletionEntry which creates a popup menu that responds to keystrokes to navigate through the items without losing the editing ability of the text input..
+func NewCompletionEntry(options []string) *CompletionEntry {
+	c := &CompletionEntry{}
+	// c.ExtendBaseWidget(c)
+	c.Entry = widget.NewEntry()
+	c.items = options
```

Comment:
You can add this to the original struct like so:
```go
c := &CompletionEntry{items: options}
```

Reply:
right too...